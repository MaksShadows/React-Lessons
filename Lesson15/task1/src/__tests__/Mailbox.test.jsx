import React from 'react'
import {shallow} from 'enzyme'
import Mailbox from '../Mailbox'


describe('<Mailbox/>', () => {
  it('should not render count if no underead messages', () => {

    const wrappedComponent = shallow(<Mailbox unreadMessages={[]}/>);

    expect(wrappedComponent.find('.mailbox__count').exists()).toEqual(false);
  });

  it('should render corect  count if no underead messages', () => {

    const unreadMessages = [1,2,3];

    const wrappedComponent = shallow(<Mailbox unreadMessages={unreadMessages}/>);

    expect(wrappedComponent.find('.mailbox__count').text()).toEqual('3');
  });

});
