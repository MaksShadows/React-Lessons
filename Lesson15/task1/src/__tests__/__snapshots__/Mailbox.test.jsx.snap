// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mailbox/> should not render count if no underead messages 1`] = `
<div
  className="mailbox"
>
  <span
    className="mailbox__text"
  >
    Messages
  </span>
</div>
`;

exports[`<Mailbox/> should return 2 unreadmessages 1`] = `
<div
  className="mailbox"
>
  <span
    className="mailbox__text"
  >
    Messages
  </span>
  <span
    className="mailbox__count"
  >
    2
  </span>
</div>
`;

exports[`<Mailbox/> should return only Mailbox without undereadmessages 1`] = `
<div
  className="mailbox"
>
  <span
    className="mailbox__text"
  >
    Messages
  </span>
</div>
`;
